src/xtopcom/xvm/xsystem_contracts/src/xtable_reward_claiming_contract.cpp

```diff
@@ -95,8 +97,151 @@ void xtop_table_reward_claiming_contract::update_vote_reward_record(common::xacc
     base::xstream_t stream(base::xcontext_t::instance());
     record.serialize_to(stream);
     auto value_str = std::string((char *)stream.data(), stream.size());
-    {
-        MAP_SET(property, account.to_string(), value_str);
+    { MAP_SET(property, account.to_string(), value_str); }
+}
+
+void xtop_table_reward_claiming_contract::get_section_voters_info(const std::string & key,
+                                                                  uint32_t table_id,
+                                                                  std::map<std::string, std::string> & voters,
+                                                                  std::map<std::string, std::map<std::string, std::string>> & pledge_votes_map,
+                                                                  std::map<std::string, uint64_t> & stored_expire_token_map) {
+    { MAP_COPY_GET(key, voters, data::xdatautil::serialize_owner_str(sys_contract_sharding_vote_addr, table_id)); }
+    xinfo("[xtop_table_reward_claiming_contract::recv_voter_dividend_reward] reward-claiming table_id:%d,vote_valid_rate vote maps %s size: %d, pid: %d",
+          table_id,
+          key.c_str(),
+          voters.size(),
+          getpid());
+
+    for (auto const & entity : voters) {
+        auto const & account = entity.first;
+        std::map<std::string, std::string> pledge_votes;
+        MAP_COPY_GET(data::XPROPERTY_PLEDGE_VOTE_KEY, pledge_votes, account);
+        if (pledge_votes.empty()) {
+            xdbg("[xtop_table_reward_claiming_contract::recv_voter_dividend_reward] reward-claiming pledge votes is empty,table_id:%d,account:%s", table_id, account);
+            // do nothing
+            continue;
+        }
+        pledge_votes_map.insert({account, pledge_votes});
+
+        std::string val = STRING_GET2(data::XPROPERTY_EXPIRE_VOTE_TOKEN_KEY, account);
+        stored_expire_token_map.insert({account, xstring_utl::touint64(val)});
+    }
+}
+
+void xtop_table_reward_claiming_contract::calc_section_votes_table_and_adv_vote(std::map<std::string, std::string> & voters,
+                                                                                std::map<std::string, std::map<std::string, std::string>> & pledge_votes_map,
```

Comment:
looks like `pledge_votes_map` isn't being modified in the function? is so, make this parameter `std::map<std::string, std::map<std::string, std::string>> const & pledge_votes_map`